--- glfw-2.7.1/compile.sh.orig	2011-06-28 05:55:28.000000000 +0200
+++ glfw-2.7.1/compile.sh	2011-07-30 18:04:01.660628707 +0200
@@ -107,35 +107,8 @@
 ##########################################################################
 echo -n "Checking for X11 libraries location... " 1>&6
 
-if [ -r "/usr/X11/lib" ]; then
-  GLFW_LFLAGS="$GLFW_LFLAGS -L/usr/X11/lib"
-  GLFW_CFLAGS="-I/usr/X11/include $GLFW_CFLAGS"
-  echo "/usr/X11/lib" 1>&6
-elif [ -r "/usr/X11R7/lib" ]; then
-  GLFW_LFLAGS="$GLFW_LFLAGS -L/usr/X11R7/lib"
-  GLFW_CFLAGS="-I/usr/X11R7/include $GLFW_CFLAGS"
-  echo "/usr/X11R7/lib" 1>&6
-elif [ -r "/usr/X11R6/lib" ]; then
-  GLFW_LFLAGS="$GLFW_LFLAGS -L/usr/X11R6/lib"
-  GLFW_CFLAGS="-I/usr/X11R6/include $GLFW_CFLAGS"
-  echo "/usr/X11R6/lib" 1>&6
-elif [ -r "/usr/X11R5/lib" ]; then
-  GLFW_LFLAGS="$GLFW_LFLAGS -L/usr/X11R5/lib"
-  GLFW_CFLAGS="-I/usr/X11R5/include $GLFW_CFLAGS"
-  echo "/usr/X11R5/lib" 1>&6
-elif [ -r "/opt/X11R6/lib" ]; then
-  # This location is used on QNX
-  GLFW_LFLAGS="$GLFW_LFLAGS -L/opt/X11R6/lib"
-  GLFW_CFLAGS="-I/opt/X11R6/include $GLFW_CFLAGS"
-  echo "/opt/X11R6/lib" 1>&6
-elif [ -r "/usr/X/lib" ]; then
-  GLFW_LFLAGS="$GLFW_LFLAGS -L/usr/X/lib"
-  GLFW_CFLAGS="-I/usr/X/include $GLFW_CFLAGS"
-  echo "/usr/X/lib" 1>&6
-else
   # TODO: Detect and report X11R7 in /usr/lib
   echo "unknown (assuming linker will find them)" 1>&6
-fi
 
 
 ##########################################################################
@@ -571,9 +544,7 @@
 
 if [ "x$CFLAGS" = x ]; then
   if [ "x$use_gcc" = xyes ]; then
-    GLFW_CFLAGS="$GLFW_CFLAGS -O2 -Wall"
-  else
-    GLFW_CFLAGS="$GLFW_CFLAGS -O"
+    GLFW_CFLAGS="$GLFW_CFLAGS -Wall"
   fi
 fi
 
