--- glfw-2.5/lib/x11/x11_time.c.orig	2005-01-01 18:49:56.000000000 +0100
+++ glfw-2.5/lib/x11/x11_time.c	2006-12-27 23:52:42.788696559 +0100
@@ -147,17 +147,17 @@
     asm(
         "pushf\n\t"
         "pop    %%eax\n\t"
-        "movl   %%eax,%%ebx\n\t"
+        "movl   %%eax,%%ecx\n\t"
         "xorl   $0x00200000,%%eax\n\t"
         "push   %%eax\n\t"
         "popf\n\t"
         "pushf\n\t"
         "pop    %%eax\n\t"
-        "xorl   %%eax,%%ebx\n\t"
+        "xorl   %%eax,%%ecx\n\t"
         "movl   %%eax,%0\n\t"
         : "=m" (has_cpuid)
         :
-        : "%eax", "%ebx"
+        : "%eax", "%ecx"
     );
     if( !has_cpuid )
     {
@@ -167,14 +167,16 @@
     // Execute CPUID
     asm(
         "movl   %4,%%eax\n\t"
+	"pushl  %%ebx\n\t"
         "cpuid\n\t"
         "movl   %%eax,%0\n\t"
         "movl   %%ebx,%1\n\t"
         "movl   %%ecx,%2\n\t"
         "movl   %%edx,%3\n\t"
+	"popl   %%ebx\n\t"
         : "=m" (local_a), "=m" (local_b), "=m" (local_c), "=m" (local_d)
         : "m" (ID)
-        : "%eax", "%ebx", "%ecx", "%edx"
+        : "%eax", "%ecx", "%edx"
     );
 
     *a = local_a;
